core/src/main/java/org/arquillian/cube/impl/client/container/CubeContainerLifecycleController.java

```diff
@@ -24,7 +24,8 @@
     @Inject
     private Event<CubeControlEvent> controlEvent;
 
-    public void startCubeMappedContainer(@Observes BeforeStart event, CubeRegistry cubeRegistry,
+    // Must be started before ContainerConfigurationController
+    public void startCubeMappedContainer(@Observes(precedence=100) BeforeSetup event, CubeRegistry cubeRegistry,
```

Comment:
Might be some code in there that is woth looking at: https://github.com/mryan43/arquillian-available-port-extension

'Finding an open port' is probably good enough. Assume it won't be taken over the next few seconds. :)